src/storage/src/source/generator.rs

```diff
@@ -245,9 +260,28 @@ fn render_simple_generator<G: Scope<Timestamp = MzOffset>>(
             return;
         };
 
-        let mut rows = generator.by_seed(mz_ore::now::SYSTEM_TIME.clone(), None, resume_offset);
-
+        let now_fn = mz_ore::now::SYSTEM_TIME.clone();
+
+        let start_instant = {
+            // We want to have our interval start at a nice round number...
+            // for example, if our tick interval is one minute, to start at a minute boundary.
+            // However, the `Interval` type from tokio can't be "floored" in that way.
+            // Instead, figure out the amount we should step back based on the wall clock,
+            // then apply that to our monotonic clock to make things start at approximately the
+            // right time.
+            let now_millis = now_fn();
+            let now_instant = Instant::now();
+            let delay_millis = tick_micros
+                .map(|tick_micros| tick_micros / 1000)
+                .filter(|tick_millis| *tick_millis > 0)
+                .map(|tick_millis| tick_millis - now_millis.rem(tick_millis))
+                .unwrap_or(0);
+            now_instant + Duration::from_millis(delay_millis)
+        };
         let tick = Duration::from_micros(tick_micros.unwrap_or(1_000_000));
+        let mut tick_interval = interval_at(start_instant, tick);
+
+        let mut rows = generator.by_seed(now_fn, None, resume_offset);
```

Comment:
This logic looks good to my relatively untrained eye, but we should get @petrosagg's eyes on it too.

Reply:
I think we can leave this section as it was if we change the generator to produce definite data as described in the other comment.

Reply:
@petrosagg  - Can you expand on that a bit?

Suppose my tick interval is `1 minute`, and my source starts up at 12:00:30. The iterator seems like it would be polled at roughly 12:01:30, 12:02:30, etc. I'd like the clock to update with the 12:02:00 at ~roughly 12:02:00, but it looks to me like this source wouldn't have the ability to yield at anything near that time...